JeMPI_Apps/JeMPI_UI/src/config/index.ts

```diff
@@ -0,0 +1,46 @@
+export type Config = {
+  isDev: boolean
+  apiUrl: string
+  shouldMockBackend: boolean
+  KeyCloakUrl: string
+  KeyCloakRealm: string
+  KeyCloakClientId: string
+  useSso: boolean
+  maxUploadCsvSize: number
+}
+
+export default async function getConfig() {
+  try {
+    const res = await fetch('/config.json')
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const conf = (await res.json()) as any
+    return {
+      isDev: conf.nodeEnv !== 'production',
+      apiUrl: conf.apiUrl,
+      shouldMockBackend: conf.shouldMockBackend,
+      KeyCloakUrl: conf.KeyCloakUrl,
+      KeyCloakRealm: conf.KeyCloakRealm,
+      KeyCloakClientId: conf.KeyCloakClientId,
+      useSso: conf.useSso,
+      maxUploadCsvSize: conf.maxUploadCsvSize
+    } as Config
+  } catch {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'Unable to fetch the config json file, either some env var are missing or Jempi UI is running in development mode (yarn start)'
+    )
+    return {
+      isDev: process.env.NODE_ENV !== 'production',
+      apiUrl:
+        process.env.REACT_APP_JEMPI_BASE_URL || 'http://localhost:50000/JeMPI',
```

Comment:
Can we not use `window.location.hostname` / `window.location.host` instead?

Reply:
Also for the keycloak default url as well

Reply:
Refactored. Please take a look at the newest updated version.